kolibri/plugins/user_auth/assets/src/views/SignInPage/NewPasswordPage.vue

```diff
@@ -33,7 +33,19 @@
         :isValid.sync="passwordIsValid"
         :shouldValidate="busy"
         @submitNewPassword="updatePassword"
+        :type="showPassword ? 'text' : 'password'"  <!-- Toggle between 'text' and 'password' -->
       />
+
+      <!-- Button to toggle password visibility -->
+      <KButton
+        appearance="basic-link"
+        style="margin-top: 8px; text-align: left;"
+        @click="togglePassword"
+        data-test="togglePasswordVisibility"
+      >
+        {{ showPassword ? 'Hide' : 'Show' }} Password
```

Comment:
In order to be sure that the message here is translated for all users, you'd need to add two new messages in the `$trs` property of the component, one each for "Hide password" and "Show password". Then you can conditionalize which message to show by referencing the keys of the objects you add to `$trs` in a call to `$tr(<the key of the message object you made>)`.

Reply:
You could pass the KButton's label to the `:text` prop instead of to the default slot as you have here. It may be cleaner to do this because we also need to put the same text on the KButton's `aria-label` attribute as well